pkg/operation/botanist/addons.go

```diff
@@ -242,26 +215,11 @@ func (b *Botanist) generateOptionalAddonsChart(_ context.Context) (*chartrendere
 	}
 
 	kubernetesDashboardEnabled := gardencorev1beta1helper.KubernetesDashboardEnabled(b.Shoot.GetInfo().Spec.Addons)
-
-	nginxIngressConfig, err := b.GenerateNginxIngressConfig()
-	if err != nil {
-		return nil, err
-	}
-	nginxIngress, err := b.InjectShootShootImages(nginxIngressConfig, images.ImageNameNginxIngressController, images.ImageNameIngressDefaultBackend)
-	if err != nil {
-		return nil, err
-	}
+	nginxIngressAddonEnabled := gardencorev1beta1helper.NginxIngressEnabled(b.Shoot.GetInfo().Spec.Addons)
 
 	return b.ShootClientSet.ChartRenderer().Render(filepath.Join(charts.Path, "shoot-addons"), "addons", metav1.NamespaceSystem, map[string]interface{}{
 		"global":               global,
 		"kubernetes-dashboard": common.GenerateAddonConfig(nil, kubernetesDashboardEnabled),
-		"nginx-ingress":        nginxIngress,
+		"nginx-ingress":        common.GenerateAddonConfig(nil, nginxIngressAddonEnabled),
```

Comment:
I'm wondering whether we still need these values. IIUC, the charts in `shoots-addons` are now deployed unconditionally even if the addons are disabled via shoot spec, or?

Reply:
I tested in my local. On disabling the `nginx-ingress addons` the charts are not getting deployed(i.e I cannot find correspoding deployment, services etc) in the shoot. Am I missing something here ?

Reply:
did you fix the namespace issue, I mentioned?

Reply:
How does this work without the `requirements.yaml` file? 👀

Reply:
The chart is deployed unconditionally, but it is always empty
```
❯ k get secrets -n shoot--local--gcp managedresource-addons -o yaml | yq -e '.data.["shoot-addons_charts_nginx-ingress_templates_controller-deployment.yaml"]' | base64 -d
apiVersion: apps/v1
kind: Deployment
metadata:
  name: addons-nginx-ingress-controller
  namespace: kube-system
  annotations:
    resources.gardener.cloud/mode: Ignore
```

Reply:
Thank you @oliver-goetz . I was checking the actual deployment instead of checking the `managedresource-addons` secret. I too checked now even after removing the statement `"nginx-ingress":        common.GenerateAddonConfig(nil, nginxIngressAddonEnabled)` , the charts are getting deployed. Hence this statement can be removed.
```
❯ k get secret -n shoot--dev--i546764-awsshoot1 managedresource-addons -oyaml
apiVersion: v1
data:
  shoot-addons_charts_kubernetes-dashboard_templates_configmap.yaml: IyBUT0RPKGFjdW1pbm8pOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCi0tLQpraW5kOiBDb25maWdNYXAKYXBpVmVyc2lvbjogdjEKbWV0YWRhdGE6CiAgbmFtZToga3ViZXJuZXRlcy1kYXNoYm9hcmQtc2V0dGluZ3MKICBuYW1lc3BhY2U6IGt1YmVybmV0ZXMtZGFzaGJvYXJkCiAgYW5ub3RhdGlvbnM6CiAgICByZXNvdXJjZXMuZ2FyZGVuZXIuY2xvdWQvbW9kZTogSWdub3JlCg==
  shoot-addons_charts_kubernetes-dashboard_templates_deployment-dashboard.yaml: IyBUT0RPKGFjdW1pbm8pOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IGFwcHMvdjEKa2luZDogRGVwbG95bWVudAptZXRhZGF0YToKICBuYW1lOiBrdWJlcm5ldGVzLWRhc2hib2FyZAogIG5hbWVzcGFjZToga3ViZXJuZXRlcy1kYXNoYm9hcmQKICBhbm5vdGF0aW9uczoKICAgIHJlc291cmNlcy5nYXJkZW5lci5jbG91ZC9tb2RlOiBJZ25vcmUK
  shoot-addons_charts_kubernetes-dashboard_templates_deployment-metrics-scraper.yaml: IyBUT0RPKGFjdW1pbm8pOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCi0tLQphcGlWZXJzaW9uOiBhcHBzL3YxCmtpbmQ6IERlcGxveW1lbnQKbWV0YWRhdGE6CiAgbmFtZTogZGFzaGJvYXJkLW1ldHJpY3Mtc2NyYXBlcgogIG5hbWVzcGFjZToga3ViZXJuZXRlcy1kYXNoYm9hcmQKICBhbm5vdGF0aW9uczoKICAgIHJlc291cmNlcy5nYXJkZW5lci5jbG91ZC9tb2RlOiBJZ25vcmUK
  shoot-addons_charts_kubernetes-dashboard_templates_namespace.yaml: IyBUT0RPKGFjdW1pbm8pOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IHYxCmtpbmQ6IE5hbWVzcGFjZQptZXRhZGF0YToKICBuYW1lOiBrdWJlcm5ldGVzLWRhc2hib2FyZAogIGFubm90YXRpb25zOgogICAgcmVzb3VyY2VzLmdhcmRlbmVyLmNsb3VkL21vZGU6IElnbm9yZQo=
  shoot-addons_charts_kubernetes-dashboard_templates_rbac.yaml: IyBUT0RPKGFjdW1pbm8pOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCi0tLQphcGlWZXJzaW9uOiByYmFjLmF1dGhvcml6YXRpb24uazhzLmlvL3YxCmtpbmQ6IFJvbGUKbWV0YWRhdGE6CiAgbmFtZToga3ViZXJuZXRlcy1kYXNoYm9hcmQKICBuYW1lc3BhY2U6IGt1YmVybmV0ZXMtZGFzaGJvYXJkCiAgYW5ub3RhdGlvbnM6CiAgICByZXNvdXJjZXMuZ2FyZGVuZXIuY2xvdWQvbW9kZTogSWdub3JlCgotLS0KYXBpVmVyc2lvbjogcmJhYy5hdXRob3JpemF0aW9uLms4cy5pby92MQpraW5kOiBSb2xlQmluZGluZwptZXRhZGF0YToKICBuYW1lOiBrdWJlcm5ldGVzLWRhc2hib2FyZAogIG5hbWVzcGFjZToga3ViZXJuZXRlcy1kYXNoYm9hcmQKICBhbm5vdGF0aW9uczoKICAgIHJlc291cmNlcy5nYXJkZW5lci5jbG91ZC9tb2RlOiBJZ25vcmUKCi0tLQphcGlWZXJzaW9uOiByYmFjLmF1dGhvcml6YXRpb24uazhzLmlvL3YxCmtpbmQ6IENsdXN0ZXJSb2xlCm1ldGFkYXRhOgogIG5hbWU6IGt1YmVybmV0ZXMtZGFzaGJvYXJkCiAgYW5ub3RhdGlvbnM6CiAgICByZXNvdXJjZXMuZ2FyZGVuZXIuY2xvdWQvbW9kZTogSWdub3JlCgotLS0KYXBpVmVyc2lvbjogcmJhYy5hdXRob3JpemF0aW9uLms4cy5pby92MQpraW5kOiBDbHVzdGVyUm9sZUJpbmRpbmcKbWV0YWRhdGE6CiAgbmFtZToga3ViZXJuZXRlcy1kYXNoYm9hcmQKICBhbm5vdGF0aW9uczoKICAgIHJlc291cmNlcy5nYXJkZW5lci5jbG91ZC9tb2RlOiBJZ25vcmUK
  shoot-addons_charts_kubernetes-dashboard_templates_secret.yaml: IyBUT0RPKGFjdW1pbm8pOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCi0tLQphcGlWZXJzaW9uOiB2MQpraW5kOiBTZWNyZXQKbWV0YWRhdGE6CiAgbmFtZToga3ViZXJuZXRlcy1kYXNoYm9hcmQtY2VydHMKICBuYW1lc3BhY2U6IGt1YmVybmV0ZXMtZGFzaGJvYXJkCiAgYW5ub3RhdGlvbnM6CiAgICByZXNvdXJjZXMuZ2FyZGVuZXIuY2xvdWQvbW9kZTogSWdub3JlCgotLS0KYXBpVmVyc2lvbjogdjEKa2luZDogU2VjcmV0Cm1ldGFkYXRhOgogIG5hbWU6IGt1YmVybmV0ZXMtZGFzaGJvYXJkLWNzcmYKICBuYW1lc3BhY2U6IGt1YmVybmV0ZXMtZGFzaGJvYXJkCiAgYW5ub3RhdGlvbnM6CiAgICByZXNvdXJjZXMuZ2FyZGVuZXIuY2xvdWQvbW9kZTogSWdub3JlCgotLS0KYXBpVmVyc2lvbjogdjEKa2luZDogU2VjcmV0Cm1ldGFkYXRhOgogIG5hbWU6IGt1YmVybmV0ZXMtZGFzaGJvYXJkLWtleS1ob2xkZXIKICBuYW1lc3BhY2U6IGt1YmVybmV0ZXMtZGFzaGJvYXJkCiAgYW5ub3RhdGlvbnM6CiAgICByZXNvdXJjZXMuZ2FyZGVuZXIuY2xvdWQvbW9kZTogSWdub3JlCg==
  shoot-addons_charts_kubernetes-dashboard_templates_service-dashboard.yaml: IyBUT0RPKGFjdW1pbm8pOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCi0tLQphcGlWZXJzaW9uOiB2MQpraW5kOiBTZXJ2aWNlCm1ldGFkYXRhOgogIG5hbWU6IGt1YmVybmV0ZXMtZGFzaGJvYXJkCiAgbmFtZXNwYWNlOiBrdWJlcm5ldGVzLWRhc2hib2FyZAogIGFubm90YXRpb25zOgogICAgcmVzb3VyY2VzLmdhcmRlbmVyLmNsb3VkL21vZGU6IElnbm9yZQo=
  shoot-addons_charts_kubernetes-dashboard_templates_service-metrics-scraper.yaml: IyBUT0RPKGFjdW1pbm8pOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCi0tLQphcGlWZXJzaW9uOiB2MQpraW5kOiBTZXJ2aWNlCm1ldGFkYXRhOgogIG5hbWU6IGRhc2hib2FyZC1tZXRyaWNzLXNjcmFwZXIKICBuYW1lc3BhY2U6IGt1YmVybmV0ZXMtZGFzaGJvYXJkCiAgYW5ub3RhdGlvbnM6CiAgICByZXNvdXJjZXMuZ2FyZGVuZXIuY2xvdWQvbW9kZTogSWdub3JlCgo=
  shoot-addons_charts_kubernetes-dashboard_templates_serviceaccount.yaml: IyBUT0RPKGFjdW1pbm8pOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCi0tLQphcGlWZXJzaW9uOiB2MQpraW5kOiBTZXJ2aWNlQWNjb3VudAptZXRhZGF0YToKICBuYW1lOiBrdWJlcm5ldGVzLWRhc2hib2FyZAogIG5hbWVzcGFjZToga3ViZXJuZXRlcy1kYXNoYm9hcmQKICBhbm5vdGF0aW9uczoKICAgIHJlc291cmNlcy5nYXJkZW5lci5jbG91ZC9tb2RlOiBJZ25vcmUK
  shoot-addons_charts_kubernetes-dashboard_templates_vpa.yaml: IyBUT0RPKGFjdW1pbm8pOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246ICJhdXRvc2NhbGluZy5rOHMuaW8vdjEiCmtpbmQ6IFZlcnRpY2FsUG9kQXV0b3NjYWxlcgptZXRhZGF0YToKICBuYW1lOiBrdWJlcm5ldGVzLWRhc2hib2FyZAogIG5hbWVzcGFjZToga3ViZXJuZXRlcy1kYXNoYm9hcmQKICBhbm5vdGF0aW9uczoKICAgIHJlc291cmNlcy5nYXJkZW5lci5jbG91ZC9tb2RlOiBJZ25vcmUK
  shoot-addons_charts_nginx-ingress_templates_clusterrole.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IHJiYWMuYXV0aG9yaXphdGlvbi5rOHMuaW8vdjEKa2luZDogQ2x1c3RlclJvbGUKbWV0YWRhdGE6CiAgbmFtZTogYWRkb25zLW5naW54LWluZ3Jlc3MKICBhbm5vdGF0aW9uczoKICAgIHJlc291cmNlcy5nYXJkZW5lci5jbG91ZC9tb2RlOiBJZ25vcmUK
  shoot-addons_charts_nginx-ingress_templates_clusterrolebinding.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IHJiYWMuYXV0aG9yaXphdGlvbi5rOHMuaW8vdjEKa2luZDogQ2x1c3RlclJvbGVCaW5kaW5nCm1ldGFkYXRhOgogIG5hbWU6IGFkZG9ucy1uZ2lueC1pbmdyZXNzCiAgYW5ub3RhdGlvbnM6CiAgICByZXNvdXJjZXMuZ2FyZGVuZXIuY2xvdWQvbW9kZTogSWdub3JlCg==
  shoot-addons_charts_nginx-ingress_templates_controller-configmap.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IHYxCmtpbmQ6IENvbmZpZ01hcAptZXRhZGF0YToKICBuYW1lOiBhZGRvbnMtbmdpbngtaW5ncmVzcy1jb250cm9sbGVyCiAgbmFtZXNwYWNlOiBrdWJlLXN5c3RlbQogIGFubm90YXRpb25zOgogICAgcmVzb3VyY2VzLmdhcmRlbmVyLmNsb3VkL21vZGU6IElnbm9yZQo=
  shoot-addons_charts_nginx-ingress_templates_controller-deployment.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IGFwcHMvdjEKa2luZDogRGVwbG95bWVudAptZXRhZGF0YToKICBuYW1lOiBhZGRvbnMtbmdpbngtaW5ncmVzcy1jb250cm9sbGVyCiAgbmFtZXNwYWNlOiBrdWJlLXN5c3RlbQogIGFubm90YXRpb25zOgogICAgcmVzb3VyY2VzLmdhcmRlbmVyLmNsb3VkL21vZGU6IElnbm9yZQo=
  shoot-addons_charts_nginx-ingress_templates_controller-service.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IHYxCmtpbmQ6IFNlcnZpY2UKbWV0YWRhdGE6CiAgbmFtZTogYWRkb25zLW5naW54LWluZ3Jlc3MtY29udHJvbGxlcgogIG5hbWVzcGFjZToga3ViZS1zeXN0ZW0KICBhbm5vdGF0aW9uczoKICAgIHJlc291cmNlcy5nYXJkZW5lci5jbG91ZC9tb2RlOiBJZ25vcmUK
  shoot-addons_charts_nginx-ingress_templates_default-backend-deployment.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IGFwcHMvdjEKa2luZDogRGVwbG95bWVudAptZXRhZGF0YToKICBuYW1lOiBhZGRvbnMtbmdpbngtaW5ncmVzcy1uZ2lueC1pbmdyZXNzLWs4cy1iYWNrZW5kCiAgbmFtZXNwYWNlOiBrdWJlLXN5c3RlbQogIGFubm90YXRpb25zOgogICAgcmVzb3VyY2VzLmdhcmRlbmVyLmNsb3VkL21vZGU6IElnbm9yZQo=
  shoot-addons_charts_nginx-ingress_templates_default-backend-service.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IHYxCmtpbmQ6IFNlcnZpY2UKbWV0YWRhdGE6CiAgbmFtZTogYWRkb25zLW5naW54LWluZ3Jlc3MtbmdpbngtaW5ncmVzcy1rOHMtYmFja2VuZAogIG5hbWVzcGFjZToga3ViZS1zeXN0ZW0KICBhbm5vdGF0aW9uczoKICAgIHJlc291cmNlcy5nYXJkZW5lci5jbG91ZC9tb2RlOiBJZ25vcmUK
  shoot-addons_charts_nginx-ingress_templates_ingressclass.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IG5ldHdvcmtpbmcuazhzLmlvL3YxCmtpbmQ6IEluZ3Jlc3NDbGFzcwptZXRhZGF0YToKICBuYW1lOiBuZ2lueAogIGFubm90YXRpb25zOgogICAgcmVzb3VyY2VzLmdhcmRlbmVyLmNsb3VkL21vZGU6IElnbm9yZQo=
  shoot-addons_charts_nginx-ingress_templates_psp_nginx-ingress-rolebinding-privileged.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IHJiYWMuYXV0aG9yaXphdGlvbi5rOHMuaW8vdjEKa2luZDogUm9sZUJpbmRpbmcKbWV0YWRhdGE6CiAgbmFtZTogZ2FyZGVuZXIuY2xvdWQ6cHNwOmFkZG9ucy1uZ2lueC1pbmdyZXNzCiAgbmFtZXNwYWNlOiBrdWJlLXN5c3RlbQogIGFubm90YXRpb25zOgogICAgcmVzb3VyY2VzLmdhcmRlbmVyLmNsb3VkL21vZGU6IElnbm9yZQoK
  shoot-addons_charts_nginx-ingress_templates_role.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IHJiYWMuYXV0aG9yaXphdGlvbi5rOHMuaW8vdjEKa2luZDogUm9sZQptZXRhZGF0YToKICBuYW1lOiBhZGRvbnMtbmdpbngtaW5ncmVzcwogIG5hbWVzcGFjZToga3ViZS1zeXN0ZW0KICBhbm5vdGF0aW9uczoKICAgIHJlc291cmNlcy5nYXJkZW5lci5jbG91ZC9tb2RlOiBJZ25vcmUK
  shoot-addons_charts_nginx-ingress_templates_rolebinding.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IHJiYWMuYXV0aG9yaXphdGlvbi5rOHMuaW8vdjEKa2luZDogUm9sZUJpbmRpbmcKbWV0YWRhdGE6CiAgbmFtZTogYWRkb25zLW5naW54LWluZ3Jlc3MKICBuYW1lc3BhY2U6IGt1YmUtc3lzdGVtCiAgYW5ub3RhdGlvbnM6CiAgICByZXNvdXJjZXMuZ2FyZGVuZXIuY2xvdWQvbW9kZTogSWdub3JlCg==
  shoot-addons_charts_nginx-ingress_templates_serviceaccount.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246IHYxCmtpbmQ6IFNlcnZpY2VBY2NvdW50Cm1ldGFkYXRhOgogIG5hbWU6IGFkZG9ucy1uZ2lueC1pbmdyZXNzCiAgbmFtZXNwYWNlOiBrdWJlLXN5c3RlbQogIGFubm90YXRpb25zOgogICAgcmVzb3VyY2VzLmdhcmRlbmVyLmNsb3VkL21vZGU6IElnbm9yZQo=
  shoot-addons_charts_nginx-ingress_templates_vpa.yaml: IyBUT0RPKGFyeTE5OTIpOiBEZWxldGUgdGhpcyBIZWxtIGNoYXJ0IGluIGEgZnV0dXJlIHZlcnNpb24uCmFwaVZlcnNpb246ICJhdXRvc2NhbGluZy5rOHMuaW8vdjEiCmtpbmQ6IFZlcnRpY2FsUG9kQXV0b3NjYWxlcgptZXRhZGF0YToKICBuYW1lOiBhZGRvbnMtbmdpbngtaW5ncmVzcy1jb250cm9sbGVyCiAgbmFtZXNwYWNlOiBrdWJlLXN5c3RlbQogIGFubm90YXRpb25zOgogICAgcmVzb3VyY2VzLmdhcmRlbmVyLmNsb3VkL21vZGU6IElnbm9yZQo=
kind: Secret
metadata:
  creationTimestamp: "2023-01-18T09:26:55Z"
  finalizers:
  - resources.gardener.cloud/gardener-resource-manager
  name: managedresource-addons
  namespace: shoot--dev--i546764-awsshoot1
  resourceVersion: "504373"
  uid: ceb0bd9c-3998-406e-90a7-ded36c5c7e8d
type: Opaque
```